examples/neighbors/plot_variables_choice.py

```diff
@@ -0,0 +1,79 @@
+"""
+=================================================================================
+Nearest Neighbors Classification with different values of n_neighbors and weights
+=================================================================================
+
+Sample usage of Nearest Neighbors classification.
+It will plot the decision boundaries for each class.
+
+This example show the effect of tweaking the parameters ``n_neighbors`` and ``weights``. 
+The smaller ``n_neighbors`` is the noisier the boundaries are.
+"""
+print(__doc__)
+
+from time import time
+from sklearn import neighbors, datasets
+from matplotlib.colors import ListedColormap
+import seaborn as sns
+import matplotlib.pyplot as plt
+import numpy as np
+
+# import some data to play with
+iris = datasets.load_iris()
+
+# we only take the first two features. We could avoid this ugly
+# slicing by using a two-dim dataset
+X = iris.data[:, :2]
+y = iris.target
+
+# Create the mesh
+h = .02  # step size in the mesh
+x_min, x_max = X[:, 0].min() - 1, X[:, 0].max() + 1
+y_min, y_max = X[:, 1].min() - 1, X[:, 1].max() + 1
+xx, yy = np.meshgrid(np.arange(x_min, x_max, h), np.arange(y_min, y_max, h))
+mesh = np.c_[xx.ravel(), yy.ravel()]
+
+# Create color maps
+cmap_light = ListedColormap(['orange', 'cyan', 'cornflowerblue'])
+cmap_bold = ['darkorange', 'c', 'darkblue']
+
+tested_parameters = {
+    "n_neighbors": [1, 5, 15, 150],
```

Comment:
Maybe the following would be more informative:

```suggestion
    "n_neighbors": [1, 5, 50, 150],
```

I would expect that with 50 we already get smooth decision boundaries and some under-fitting for both weighting schemes. The case where `n_neighbors == 150 == n_samples` case is interesting but is an edge case (in particular for uniform weights).